lib/instance_billing_flavor_check/utils.py

```diff
@@ -240,25 +242,32 @@ def make_request(rmt_ip_addr, metadata, identifier):
             message = 'Request error:{}'.format(err)
         except Exception as err:
             message = 'Unexpected error: {}'.format(err)
-    
+
         if message:
-            logger.warning(
-                'Attempt {}: failed: {}'.format(retry_count, message)
-            )
-            if 'Timeout' in message:
-                retry_count += 1
+            if 'Timeout' in message or 'Connecting' in message:
+                logger.warning(
+                    'Attempt {}: failed: {}'.format(retry_count, message)
+                )
                 time.sleep(2)
-            if 'Timeout' in message or retry_count == 4:
-                return
-
-        if response.status_code == 200:
-            result = response.json()
-            logger.debug(result)
-            return result.get('flavor')
-
-    logger.error(
-        'Request to check if instance is PAYG/BYOS failed: %s', response.reason
-    )
+                retry_count += 1
+                continue
+            else:
+                # error is not time out => return None
+                logger.error(
+                    'Request to check if instance is PAYG/BYOS failed: %s',
+                    message
+                )
```

Comment:
We need a `return` here

Reply:
And we need a `break` or `retry_count = 4`  to get out of the while loop. Now we have the same endless loop that is being attempted to fix ;)

Reply:
no, we do not

I dont if GitHub is not showing the diff properly but basically, we have

```python
while counter is not 4
     do_stuff()
    
    if message:
        if we want to continue trying:
             update counter
             continue
        else:
             log foo
    else:
        if OK:
            get flavor
        else:
            log bar
    
    return flavor or None
```

Reply:
> Then the comment has to change.

do you mean the log message or the actual comment starting with `#`  ?

Reply:
comment updated to include both errors (timeout and connection)